model/process_text_spm.go

```diff
@@ -0,0 +1,220 @@
+package model
+
+import (
+	"iter"
+	"log/slog"
+	"strings"
+
+	"github.com/dlclark/regexp2"
+	queue "github.com/emirpasic/gods/queues/priorityqueue"
```

Comment:
The Go standard library has an example of how to build a priority queue with a heap. Should we try that instead of the dependency?
https://pkg.go.dev/container/heap